### ISSUE TITLE
Binary Websocket sends data as text instead of binary

### ISSUE THREAD

**Issue by Ne00n** [xtermjs/xterm.js#4555]
I am connecting to a websocket, which only accepts and sends binary data.

xterm does parse incoming binary data however even if I set the websocket which I attached with attachAddon, to binaryTpe = "arraybuffer" it still continues to send as string/text instead of binary.
Neither do I have found an option to manually override this behaviour.

## Details
- Browser and browser version: Chrome 114
- OS version: Debian 11
- xterm.js version: 5.2.1

### Steps to reproduce

1. create websocket and set binaryType to arraybuffer
2. attach websocket to AttachAddon
3. loadAddon attachAddon
4. write stuff into the term, which will be send as string

Anyone got an idea on this?

**Comment by jerch** [comment]
Yes client --> server way always sends text input as text (everything from `term.onData` event). This is needed to distingish text input from binary data as sent by `term.onBinary`.
If you want both to be sent with binary transport, you'd have to implement a protocol, that still makes it possible to separate both event types, if you want to able to utf-8 encode the text parts on server side. If you dont care for utf-8 encoding on server side, you can simply send both event types as Uint8Array (thus always getting bytes, but ofc lose the ability to safely convert text back into unicode).
